lib/openglvis.cpp

```diff
@@ -137,6 +140,113 @@ VisualizationScene::VisualizationScene()
 
 VisualizationScene::~VisualizationScene() {}
 
+
+void VisualizationScene
+:: DrawCutTriangle(gl3::GlDrawable& buff,
+                   const double (&p)[4][3], const double (&cv)[4],
+                   const double minv, const double maxv)
+{
+   // element center
+   double c[3];
+   c[0] = c[1] = c[2] = 0.0;
+   for (int j = 0; j < 3; j++)
+   {
+      c[0] += p[j][0]; c[1] += p[j][1]; c[2] += p[j][2];
+   }
+   c[0] /= 3.0; c[1] /= 3.0; c[2] /= 3.0;
+
+   double l = cut_lambda;
+   double q[3][3];
+
+   for (int j = 0; j < 3; j++)
+   {
+      q[j][0] = l*p[j][0] + (1-l)*c[0];
```

Comment:
I would prefer to see `1.+l` or `1.0+l` and avoid the implicit type conversion.

Reply:
ok

Reply:
Thanks